src/main/java/frc/robot/auto/DriveSegment.java

```diff
@@ -0,0 +1,104 @@
+package frc.robot.auto;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.Constants;
+import frc.robot.subsystems.Drive;
+import frc.robot.util.DriveSignal;
+
+/** A type of {@link Segment} for driving straight. */
+public class DriveSegment extends Segment {
+  private final int acceptableError = 50;
+
+  private double feet; // For cloning
+  private int cruise;
+  private int accel;
+  private boolean test;
+
+  private int targetPosition;
+  private int min, max;
+
+  private boolean done = false;
+  private Drive drive;
+
+  // This allows us to only log the encoder positions once every five ticks
+  private int loggingCount = 0;
+
+  // The number of ticks for which we've been within the acceptable range
+  private int debounceCount = 0;
+
+  public DriveSegment(double feet, int cruise, int accel) {
+    this.feet = feet;
+    this.drive = Drive.getInstance();
+    this.targetPosition = drive.feetToTicks(feet);
+    this.min = targetPosition - acceptableError;
+    this.max = targetPosition + acceptableError;
+
+    this.cruise = cruise;
+    this.accel = accel;
+    this.test = true;
+  }
+
+  @Override
+  public void init() {
+    logger.info("Initializing drive segment");
+    logger.info("Target: " + targetPosition);
+
+    drive.zeroEncoders();
+    drive.setCruiseAndAcceleration(cruise, accel);
+
+    if (test) logger.info("Hello!");
+    logger.info("Cruise: " + cruise);
+    logger.info("Accel: " + accel);
+
+    drive.setMotionMagicTarget(targetPosition, targetPosition);
+  }
+
+  @Override
+  public void tick() {
+
+    int left = drive.getLeftEncoderDistance();
+    int right = -drive.getRightEncoderDistance();
+
+    if (++loggingCount > 5) {
+      logger.info(
+          "Encoder distances: ("
+              + drive.getLeftEncoderDistance()
+              + ", "
+              + -drive.getRightEncoderDistance()
+              + ")");
+      loggingCount = 0;
+    }
+
+    // SmartDashboard.putNumber("left encoder", left);
+    // SmartDashboard.putNumber("right encoder", right);
```

Comment:
Should we put these on the dashboard while in test perhaps?

Reply:
We're already logging them, and the drive team doesn't really need to know them.